app/src/main/java/com/cornellappdev/scoop/ui/components/post/FirstPage.kt

```diff
@@ -0,0 +1,202 @@
+                                proceedEnabled = false
+                                coroutineScope.launch {
+                                    disableMessage()
+                                }
+                            }
+                            departureLocation.value.isEmpty() -> {
+                                showInvalidDepartureMessage = true
+                                proceedEnabled = false
+                                coroutineScope.launch {
+                                    disableMessage()
+                                }
+                            }
+                            arrivalLocation.value.isEmpty() -> {
+                                showInvalidArrivalMessage = true
+                                proceedEnabled = false
+                                coroutineScope.launch {
+                                    disableMessage()
+                                }
+                            }
+                            else -> {
+                                onProceedClicked()
+                            }
+                        }
+                    },
+                    contentPadding = PaddingValues(10.dp),
+                    colors = buttonColors(backgroundColor = Gray)
+                ) {
+                    Icon(
+                        Icons.Outlined.ArrowForward,
+                        contentDescription = stringResource(R.string.arrow_forward_description)
+                    )
+                }
+            }
+        }
+
+        Box {
+            BuildMessage(
+                showInvalidMethodMessage,
+                stringResource(R.string.invalid_method_of_transportation)
+            )
+            BuildMessage(showInvalidDepartureMessage, stringResource(R.string.invalid_departure))
+            BuildMessage(showInvalidArrivalMessage, stringResource(R.string.invalid_arrival))
+        }
+    }
+}
+
+@Composable
+fun TextFields(
+    methodOfTransportation: MutableState<String>,
+    departureLocation: MutableState<String>,
+    arrivalLocation: MutableState<String>,
+    tripState: MutableState<Trip>
+) {
+    Column(
+        modifier = Modifier
+            .background(color = Color.White)
+    ) {
+        Text(
+            text = stringResource(R.string.method_of_transportation),
+            modifier = Modifier
+                .fillMaxWidth(),
+            fontSize = 22.sp
+        )
+        Spinner(methodOfTransportation, listOf("Car", "Uber", "Train", "Bicycle")) {
+            tripState.value = tripState.value.copy(
+                methodOfTransportation = it
+            )
+        }
+        Spacer(modifier = Modifier.height(30.dp))
+        Text(
+            text = stringResource(R.string.locations),
+            fontSize = 22.sp
+        )
+        Spacer(modifier = Modifier.height(14.dp))
+        Row {
+            Icon(
+                imageVector = Icons.Outlined.NearMe,
+                contentDescription = null,
```

Comment:
Some of these contentDescriptions are null some place and filled out in another; just fill them out for consistency and accessibility.